packages/dbml-core/types/parse/error.d.ts

```diff
@@ -0,0 +1,29 @@
+export declare type WarningLevel =
+  | 'error'
+  | 'warning'
+  | 'info';
+
+export declare type ErrorCode = number;
+
+export declare interface EditorPosition {
+  line: number;
+  column: number;
+}
+
+export interface CompilerDiagnostic {
+  readonly message: Readonly<string>;
+  readonly filepath?: Readonly<string>;
+  readonly location: {
+      start: Readonly<EditorPosition>,
+      // in monaco, if an end position is not specified
+      // it consumes the word containing the start position
+      end?: Readonly<EditorPosition>,
+  };
+  readonly type?: Readonly<WarningLevel>;
+  readonly code?: Readonly<ErrorCode>; // the error code
+  readonly [key: string]: any;
```

Comment:
@NQPhuc It allows a diagnostic to carry additional field when you define an object literal anh
```
   const diagnostic: CompilerDiagnostic = {
       message:
       location:
       somefield: ...
   } // error
```

Reply:
Actually, I defined this because I was using Jsdoc annotations + tsconfig to enforce conformation to the CompilerDiagnostic and some object literal contains additional field so Typescript was complaining.
But I decided to remove them all together so this is now redundant.